Doc/library/math.rst

```diff
@@ -284,6 +286,30 @@ Number-theoretic and representation functions
    :class:`~numbers.Integral` (usually an integer). Delegates to
    :meth:`x.__trunc__() <object.__trunc__>`.
 
+.. function:: ulp(x)
+
+   Return the value of the least significant bit of the float *x*:
+
+   * If *x* is a NaN (not a number), return *x*.
+   * If *x* is negative, return ``ulp(-x)``.
+   * If *x* is a positive infinity, return *x*.
+   * If *x* is equal to positive zero, return the smallest positive
```

Comment:
I would remove "positive", because it is not clear what happens for `-0.0` (it is not negative in common meaning).

Reply:
Agreed: "equal to zero" is both technically accurate and unlikely to cause confusion here.

Reply:
I wanted to explicitly mention +0.0 to distinguish it from -0.0... but I checked, in practice ulp(+0.0) is equal to ulp(-0.0), so it doesn't really matter to specify the zero sign :-)